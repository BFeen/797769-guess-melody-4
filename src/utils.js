export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const noOp = () => {
  // nothing to do
};
