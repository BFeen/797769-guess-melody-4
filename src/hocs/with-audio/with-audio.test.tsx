import * as React from "react";
import * as renderer from "react-test-renderer";
import withAudio from "./with-audio";
import {noOp} from "../../utils";


interface mockProps {
  children: React.ReactNode;
}

const MockComponent = (props: mockProps) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

const MockComponentWrapped = withAudio(MockComponent);

it(`withAudio snapshot testing`, () => {
  const tree = renderer.create(
      <MockComponentWrapped
        isPlaying={false}
        onPlayButtonClick={noOp}
        src={``}
      />, {
        createNodeMock() {
          return {};
        }
      }
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
