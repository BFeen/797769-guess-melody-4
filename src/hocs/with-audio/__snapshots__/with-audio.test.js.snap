// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withAudio snapshot testing 1`] = `
<div>
  <audio />
</div>
`;
